src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -1894,5 +1882,232 @@ struct S
                 //         _ = this[r]; // 3, 4
                 Diagnostic(ErrorCode.WRN_ImplicitCopyInReadOnlyMember, "this").WithArguments("S.Slice(int, int)", "this").WithLocation(12, 13));
         }
+
+        [Fact]
+        public void Subsumption_01()
```

Comment:
:memo: To assist with code review, here's a dump of the decision DAG for `Subsumption_01`:

```
State 0
    1. [case [..,42]:] AND(?DagNonNullTest(t0), t1=DagPropertyEvaluation(t0.Length), ?DagRelationalTest(t1 >= ConstantValueOne(1: Int32)), t2=DagIndexerEvaluation(t0[-1]), ?DagValueTest(t2 == ConstantValueI32(42: Int32)))
    2. [case [42]:] AND(?DagNonNullTest(t0), t1=DagPropertyEvaluation(t0.Length), ?DagValueTest(t1 == ConstantValueOne(1: Int32)), t3=DagIndexerEvaluation(t0[0]), ?DagValueTest(t3 == ConstantValueI32(42: Int32)))
    Test: ?DagNonNullTest(t0)
    TrueBranch: 1
    FalseBranch: 6
State 1
    1. [case [..,42]:] AND(t1=DagPropertyEvaluation(t0.Length), ?DagRelationalTest(t1 >= ConstantValueOne(1: Int32)), t2=DagIndexerEvaluation(t0[-1]), ?DagValueTest(t2 == ConstantValueI32(42: Int32)))
    2. [case [42]:] AND(t1=DagPropertyEvaluation(t0.Length), ?DagValueTest(t1 == ConstantValueOne(1: Int32)), t3=DagIndexerEvaluation(t0[0]), ?DagValueTest(t3 == ConstantValueI32(42: Int32)))
    Test: t1=DagPropertyEvaluation(t0.Length)
    TrueBranch: 2
State 2
    1. [case [..,42]:] AND(?DagRelationalTest(t1 >= ConstantValueOne(1: Int32)), t2=DagIndexerEvaluation(t0[-1]), ?DagValueTest(t2 == ConstantValueI32(42: Int32)))
    2. [case [42]:] AND(?DagValueTest(t1 == ConstantValueOne(1: Int32)), t3=DagIndexerEvaluation(t0[0]), ?DagValueTest(t3 == ConstantValueI32(42: Int32)))
    Test: ?DagRelationalTest(t1 >= ConstantValueOne(1: Int32))
    TrueBranch: 3
    FalseBranch: 6
State 3 REMAINING t1:[1..2147483647]
    1. [case [..,42]:] AND(t2=DagIndexerEvaluation(t0[-1]), ?DagValueTest(t2 == ConstantValueI32(42: Int32)))
    2. [case [42]:] AND(?DagValueTest(t1 == ConstantValueOne(1: Int32)), t3=DagIndexerEvaluation(t0[0]), ?DagValueTest(t3 == ConstantValueI32(42: Int32)))
    Test: t2=DagIndexerEvaluation(t0[-1])
    TrueBranch: 4
State 4 REMAINING t1:[1..2147483647]
    1. [case [..,42]:] ?DagValueTest(t2 == ConstantValueI32(42: Int32))
    2. [case [42]:] AND(?DagValueTest(t1 == ConstantValueOne(1: Int32)), t3=DagIndexerEvaluation(t0[0]), ?DagValueTest(t3 == ConstantValueI32(42: Int32)))
    Test: ?DagValueTest(t2 == ConstantValueI32(42: Int32))
    TrueBranch: 5
    FalseBranch: 6
*State 5 REMAINING t1:[1..2147483647] t2:[42..42]
    1. [case [..,42]:] MATCH
*State 6 FAIL
```
```